### PR TITLE
ADS: Line items 1/3 - Stacked PR

### PR DESCRIPTION
**Pull request by malmstein**
Task/Issue URL:  https://app.asana.com/0/1174433894299346/1202861107912489

### Description
First PR for the new Line Components. This one is for the One Line Item.
It also replaces existing items (SettingsOption) since they are now duplicate.

### Steps to test this PR
Smoke tests

### UI changes
Design specs haven't changed yet, that comes later.

### CODE REVIEW

File: app-tracking-protection/vpn-impl/src/main/res/layout/activity_device_shield_activity.xml [hunk 1]
```diff
@@ -117,13 +117,12 @@
...
-            <com.duckduckgo.mobile.android.ui.view.SettingsOption
+            <com.duckduckgo.mobile.android.ui.view.listitem.OneLineListItem
                 android:id="@+id/cta_show_all"
                 android:layout_width="match_parent"
-                android:layout_height="wrap_content"
-                android:paddingBottom="@dimen/keyline_4"
+                android:layout_height="@dimen/oneLinelistItemHeight"
+                android:layout_marginBottom="@dimen/keyline_4"
                 android:paddingStart="72dp"
-                android:text="@string/atp_ActivityCtaShowAll"
```
Comment (anikiki) [inline app-tracking-protection/vpn-impl/src/main/res/layout/activity_device_shield_activity.xml:126]: I think this was missed:
`app:primaryText="@string/atp_ActivityCtaShowAll"`

File: app-tracking-protection/vpn-impl/src/main/res/layout/activity_device_shield_activity.xml [hunk 2]
```diff
@@ -139,23 +138,23 @@
                 android:paddingStart="@dimen/keyline_4"
                 android:text="@string/atp_ActivityAbout" />
 
-            <com.duckduckgo.mobile.android.ui.view.SettingsOption
+            <com.duckduckgo.mobile.android.ui.view.listitem.OneLineListItem
                 android:id="@+id/cta_what_are_app_trackers"
                 android:layout_width="match_parent"
-                android:layout_height="wrap_content"
-                android:text="@string/atp_ActivityAppTrackersFAQ"/>
+                android:layout_height="@dimen/oneLinelistItemHeight"
```
Comment (anikiki) [inline app-tracking-protection/vpn-impl/src/main/res/layout/activity_device_shield_activity.xml:144]: I see some `OneLineListItem` have `android:layout_height="@dimen/oneLinelistItemHeight"` and some have `android:layout_height="wrap_content"`. What is the rule for it?

File: app-tracking-protection/vpn-internal/src/main/res/layout/activity_vpn_internal_settings.xml [hunk 3]
```diff
@@ -149,28 +149,25 @@
                 android:theme="@style/SettingsSwitchTheme"
             />
 
-            <TextView
+            <com.duckduckgo.mobile.android.ui.view.listitem.OneLineListItem
```
Comment (anikiki) [inline app-tracking-protection/vpn-internal/src/main/res/layout/activity_vpn_internal_settings.xml:152]: No width/height defined on any `OneLineListItem` in this xml, app crashes:
```
                android:layout_width="match_parent"
                android:layout_height="wrap_content"
```

Comment (anikiki) [reply app-tracking-protection/vpn-internal/src/main/res/layout/activity_vpn_internal_settings.xml:152]: The style for the element with id `apptp_bugreport` (Generate AppTP bug report) looks odd. 

generate_bug_report

File: app-tracking-protection/vpn-internal/src/main/res/layout/settings_option_with_loading_state.xml [hunk 4]
```diff
@@ -24,7 +24,7 @@
 
     <TextView
         android:id="@+id/title"
-        style="@style/SettingsItemClickable"
+        android:background="?attr/selectableItemBackground"
```
Comment (anikiki) [inline app-tracking-protection/vpn-internal/src/main/res/layout/settings_option_with_loading_state.xml:27]: This item lost the previous styling from `<style name="SettingsItem">`.

File: app/lint-baseline.xml [hunk 5]
```diff
@@ -4081,7 +4081,7 @@
     <issue
         id="UnusedResources"
         message="The resource `R.layout.view_whitelist_entry` appears to be unused"
-        errorLine1="&lt;com.duckduckgo.mobile.android.ui.view.SingleLineListItem xmlns:android=&quot;http://schemas.android.com/apk/res/android&quot;"
+        errorLine1="&lt;com.duckduckgo.mobile.android.ui.view.listitem.SingleLineListItem xmlns:android=&quot;http://schemas.android.com/apk/res/android&quot;"
```
Comment (anikiki) [inline app/lint-baseline.xml:4084]: Do we still need this / anything related to view_whitelist_entry.xml?
`view_whitelist_entry.xml` was removed.

File: app/src/main/java/com/duckduckgo/app/settings/SettingsActivity.kt [hunk 6]
```diff
@@ -157,39 +157,38 @@ class SettingsActivity :
...
+            accessibilitySetting.setClickListener { viewModel.onAccessibilitySettingClicked() }
         }
 
         with(viewsPrivacy) {
             globalPrivacyControlSetting.setOnClickListener { viewModel.onGlobalPrivacyControlClicked() }
             autoconsentSetting.setOnClickListener { viewModel.onAutoconsentClicked() }
-            fireproofWebsites.setOnClickListener { viewModel.onFireproofWebsitesClicked() }
+            fireproofWebsites.setClickListener { viewModel.onFireproofWebsitesClicked() }
             automaticallyClearWhatSetting.setOnClickListener { viewModel.onAutomaticallyClearWhatClicked() }
-            automaticallyClearWhenSetting.setOnClickListener { viewModel.onAutomaticallyClearWhenClicked() }
```
Comment (anikiki) [inline app/src/main/java/com/duckduckgo/app/settings/SettingsActivity.kt:179]: This was missed:
```
automaticallyClearWhenSetting.setOnClickListener { viewModel.onAutomaticallyClearWhenClicked() }
```

File: common-ui/src/main/res/layout/view_one_line_list_item.xml [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+                                                   android:layout_width="match_parent"
+                                                   android:layout_height="wrap_content">
+
+    <ImageView
+        android:id="@+id/leadingIcon"
+        android:layout_width="@dimen/listItemImageContainerSize"
+        android:layout_height="@dimen/listItemImageContainerSize"
+        android:scaleType="center"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintStart_toStartOf="parent" app:layout_constraintTop_toTopOf="parent"
```
Comment (anikiki) [inline common-ui/src/main/res/layout/view_one_line_list_item.xml:31]: This layout needs to be formatted. I see some attributes on the same line.

File: app/lint-baseline.xml (hunk 5 above)
Comment (malmstein) [reply app/lint-baseline.xml:4084]: I need to update the baseline, for now it doesn't affect anything

**Review by anikiki — APPROVED** [review]
Looks good and works as expected! :tada:

File: app-tracking-protection/vpn-internal/src/main/res/layout/activity_vpn_internal_settings.xml [hunk 8]
```diff
@@ -71,49 +71,52 @@
...
+                android:id="@+id/vpnConnectivityChecksToggle"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                app:primaryText="VPN connectivity checks"
+                app:showSwitch="true"
+            />
+
+            <com.duckduckgo.mobile.android.ui.view.listitem.OneLineListItem
+                android:id="@+id/vpn_new_networking_layer_toggle"
```
Comment (anikiki) [inline app-tracking-protection/vpn-internal/src/main/res/layout/activity_vpn_internal_settings.xml:115]: The `vpn_new_networking_layer_toggle` toggle is not needed anymore. It was removed in https://github.com/duckduckgo/Android/pull/2321/files#diff-f4d229785908cc736fd3edd2d6741744ae40a39a4402cded5753ea51ae4f1f3aL120

File: app/src/main/java/com/duckduckgo/app/settings/SettingsActivity.kt (hunk 6 above)
Comment (anikiki) [reply app/src/main/java/com/duckduckgo/app/settings/SettingsActivity.kt:179]: @malmstein I think the above comment got lost :) There is a missing click listener:
`automaticallyClearWhenSetting.setOnClickListener { viewModel.onAutomaticallyClearWhenClicked() }`

**Pull request merged**